apache/submarine
# SUBMARINE-1204. Notebook names cannot start with "submarine"

### What is this PR for?
The name of the notebook cannot start with "submarine". Otherwise, the service name will conflict with the other services that the operator created.
For example, if the name of the notebook is "submarine-server", then it will conflict with the service "submarine-server" that the operator created.
This PR aims to avoid this problem by adding the notebookId in front of notebookName at k8sSubmitter, but the notebookId will not display on frontend.

### What type of PR is it?
[Bug Fix]

### Todos

### What is the Jira issue?
https://issues.apache.org/jira/projects/SUBMARINE/issues/SUBMARINE-1204?filter=allopenissues
### How should this be tested?
Run the submarine service, and check `kubectl get notebook -n submarine-user-test`, if the notebook name in k8s is an <id>-<name> format, and the frontend only displays notebook name, then it is correct.
### Screenshots (if appropriate)

### Questions:
* Do the license files need updating? No
* Are there breaking changes for older versions? No
* Does this need new documentation? No
